src/NUnitEngine/nunit.engine/Runners/MasterTestRunner.cs

```diff
@@ -196,7 +196,7 @@ XmlNode ITestRunner.Run(ITestEventListener listener, TestFilter filter)
         ITestRun ITestRunner.RunAsync(ITestEventListener listener, TestFilter filter)
         {
             var testRun = new TestRun(this);
-            testRun.Start(listener, filter);
+            testRun.RunAsync(listener, filter);
             return testRun;
```

Comment:
If `TestRun` and `AsyncTestEngineResult` were combined, this could just return the result directly. 

 ```cs
ITestRun ITestRunner.RunAsync(ITestEventListener listener, TestFilter filter)
{
      return RunAsync(listener, filter);
}
```

Is there a reason I shouldn't do this? Is there a reason `RunTestsAsynchronously` was void previously?